src/sentry/utils/http.py

```diff
@@ -104,7 +104,7 @@ def parse_uri_match(value):
     # idna/punycode encoded representation for normalization.
     if isinstance(domain, six.binary_type):
         domain = domain.decode("utf8")
-    domain = domain.encode("idna")
+    domain = domain.encode("idna").decode("utf-8")
```

Comment:
I'm not sure this is correct. Should we just be decoding `idna`?

```
In [23]: u'www.Alliancefrançaise.nu'
Out[23]: u'www.Alliancefran\xe7aise.nu'

In [24]: u'www.Alliancefrançaise.nu'.encode('idna')
Out[24]: 'www.xn--alliancefranaise-npb.nu'

In [25]: u'www.Alliancefrançaise.nu'.encode('idna').decode('utf-8'), type(u'www.Alliancefrançaise.nu'.encode('idna').decode('utf-8'))
Out[25]: (u'www.xn--alliancefranaise-npb.nu', unicode)

In [26]: u'www.Alliancefrançaise.nu'.encode('idna').decode('idna'), type(u'www.Alliancefrançaise.nu'.encode('idna').decode('idna'))
Out[26]: (u'www.alliancefran\xe7aise.nu', unicode)
```

Reply:
Not totally sure, but I think we only encode to `idna` so that we'll throw a unicode error if we fail? or is there a reason it's important to compare doma

Reply:
After looking into this it seems fine, we can just compare as idna. decoding as `unicode` is fine, since idna is just ascii compatible, it shouldn't materially change the comparison.